plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -350,6 +368,28 @@ public static function render_helper_output() {
 		return;
 	}
 
+	/**
+	 * Add tracking parameters to buttons (Renew, Purchase, etc.) on subscriptions page
+	 *
+	 * @param string $url URL to product page or to https://woocommerce.com/my-account/my-subscriptions/
+	 * @param string $utm_content value of utm_content query parameter used for tracking
+	 *
+	 * @return string URL including utm parameters for tracking
+	 */
+	public static function add_utm_params_to_url_for_subscription_link( $url, $utm_content ) {
+		$utm_params = 'utm_source=subscriptionsscreen&' .
+					  'utm_medium=product&' .
+					  'utm_campaign=wcaddons&' .
+					  'utm_content=' . $utm_content;
+
+		// there are already some URL parameters
+		if ( strpos( $url, '?' ) ) {
```

Comment:
Note that this won't handle `#` in URL, but it should be fine since we are building the URL ourselves, and server URLs don't utilize `#` seperator anyway.

Reply:
True, but as you mentioned we should be fine without supporting `#` in URL